### PR TITLE
GB-1713 - CLI error propagation

### PR DESCRIPTION
**Pull request by yoav-lavi**
# Description

## Features

- Adds error propagation for user errors and internal bugs.
  - Requires PR #1034 in the worker

# Type of change

- [ ] 💔 Breaking
- [x] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: cli/crates/cli/tests/dev.rs [hunk 1]
```diff
@@ -1,75 +1,33 @@
...
-        .unwrap();
+    let first_todo_id: String = dot_get!(todo_list, "todos.0.id");
 
-    let todo_list = response.data.todo_list_collection.edges.first().unwrap().node.clone();
-
-    assert!(todo_list.id.starts_with("TodoList#"));
-    assert!(todo_list.todos.first().unwrap().id.starts_with("Todo#"));
-
-    kill_with_children(*command.pids().first().unwrap());
+    assert!(todo_list_id.starts_with("TodoList#"));
```
Comment (Miaxos) [inline cli/crates/cli/tests/dev.rs]:
```suggestion
    assert!(todo_list_id.starts_with("todolist_"));
```

File: cli/crates/cli/tests/relations.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    let response = client.gql::<Value>(json!({ "query": RELATIONS_QUERY }).to_string());
+
+    let blog: Value = dot_get!(response, "data.blogCollection.edges.0.node");
+    let blog_id: String = dot_get!(blog, "id");
+    let first_author_id: String = dot_get!(blog, "authors.0.id");
+    let first_author_name: String = dot_get!(blog, "authors.0.name");
+    let first_authors_blogs: Vec<Value> = dot_get!(response, "data.blogCollection.edges.0.node.authors.0.blogs");
+
+    assert!(blog_id.starts_with("Blog#"));
```
Comment (Miaxos) [inline cli/crates/cli/tests/relations.rs]: ditto

File: cli/crates/server/src/bridge/sqlite.rs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#[allow(clippy::doc_markdown)] // false positive
+/// SQLite error codes as defined by <https://www.sqlite.org/rescode.html>
+///
+/// SQLite extended error codes are normally `i32`, but due to [`sqlx`] returning them as `Cow<'_, &str>` the codes here are defined as
+/// `&str`s for convenience.
+pub mod extended_error_codes {
```
Comment (Miaxos) [inline cli/crates/server/src/bridge/sqlite.rs:6]: https://github.com/0xSiO/sqlstate-rs ?

File: cli/crates/cli/tests/dev.rs (hunk 1 above)
Comment (yoav-lavi) [reply cli/crates/cli/tests/dev.rs]: @Miaxos just fixed all these, this was before testing with the new version

File: cli/crates/server/src/bridge/sqlite.rs (hunk 3 above)
Comment (yoav-lavi) [reply cli/crates/server/src/bridge/sqlite.rs:6]: SQLite error codes are a bit different than SQLState as far as I'm aware

**Review by Miaxos — APPROVED** [review]

**Pull request merged**